src/zjs_timers.c

```diff
@@ -131,7 +140,7 @@ static jerry_value_t add_timer_helper(const jerry_value_t function_obj,
     zjs_timer_t* handle = add_timer(interval, callback, this, repeat, argv, argc - 2);
     if (handle->callback_id == -1)
         return zjs_error("native_set_interval_handler: timer alloc failed");
-    jerry_set_object_native_handle(timer_obj, (uintptr_t)handle, zjs_timer_free_cb);
+    jerry_set_object_native_handle(timer_obj, (uintptr_t)handle, NULL);
 
     return timer_obj;
```

Comment:
Ah shoot this is what we're doing wrong. We're not supposed to return an object, we're supposed to return an ID number. So it's okay for that ID to get "GC'd", that doesn't make the timer go away.

Instead of creating an object above we should be assigning a unique number: 1, 2, 3... I guess to be really nice we would reset it back to 1 again after cleanup.

I guess your call whether you want to try to solve that now or we can follow up with a later patch.